ethcontract/Cargo.toml

```
@@ -17,20 +17,23 @@ to Ethereum smart contracts.
 name = "ethcontract"
 
 [features]
-default = ["derive", "http-tls", "ws-tls-tokio", "derive-http"]
+aws-kms = ["aws-sdk-kms"]
+default = ["derive", "http-tls", "ws-tls-tokio", "derive-http", "aws-kms"]
 derive = ["ethcontract-derive"]
 derive-http = ["ethcontract-derive/http"]
 http = ["web3/http"]
-http-tls = ["http", "web3/http-tls"]
 http-native-tls = ["http", "web3/http-native-tls"]
 http-rustls-tls = ["http", "web3/http-rustls-tls"]
-ws-tokio = ["web3/ws-tokio"]
+http-tls = ["http", "web3/http-tls"]
+ipc-tokio = ["web3/ipc-tokio"]
 ws-async-std = ["web3/ws-async-std"]
-ws-tls-tokio = ["web3/ws-tls-tokio"]
 ws-tls-async-std = ["web3/ws-tls-async-std"]
-ipc-tokio = ["web3/ipc-tokio"]
+ws-tls-tokio = ["web3/ws-tls-tokio"]
+ws-tokio = ["web3/ws-tokio"]
 
 [dependencies]
+asn1 = "0.15"
```

This is optional and tied to the `aws-kms` feature right?

```suggestion
asn1 = { version = "0.15", optional = true }
```